src/main/java/org/spongepowered/api/event/lifecycle/RegisterRegistryValueEvent.java

```diff
@@ -27,23 +27,93 @@
 import org.spongepowered.api.Engine;
 import org.spongepowered.api.ResourceKey;
 import org.spongepowered.api.event.GenericEvent;
-import org.spongepowered.api.registry.DefaultedRegistryValue;
+import org.spongepowered.api.registry.Registry;
+import org.spongepowered.api.registry.RegistryHolder;
+import org.spongepowered.api.registry.RegistryRegistrationSet;
 import org.spongepowered.api.registry.RegistryType;
 import org.spongepowered.eventgen.annotations.NoFactoryMethod;
 
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/**
+ * Lifecycle event for registering entries to a {@link Registry}.
+ *
+ * <p>This event may be called multiple times for the same layer
+ * as the registry is being appended and as dependencies become
+ * available. Plugins should only execute code inside the consumers.</p>
+ *
+ * <p><strong>Note:</strong> Layers might be reloadable!</p>
```

Comment:
This would mean that the consumers can potentially be called multiple times, correct? I think this should be further elaborated on.